library/core/src/task/yield_now.rs

```diff
@@ -0,0 +1,58 @@
+use crate::future::Future;
+use crate::pin::Pin;
+use crate::task::{Context, Poll};
+
+/// Cooperatively gives up a timeslice to the executor.
+///
+/// Calling this function calls move the currently executing future to the back
+/// of the execution queue, making room for other futures to execute. This is
```

Comment:
Calling this function doesn’t do anything, awaiting the returned future does. The rest of this sentence sounds like a definitive guarantee whereas the comment in the code says "most executors" which is a much weaker.

This doc should probably be rephrased along the lines of:

> Awaiting the future returned by this function yields control back to the executor, giving it an opportunity to run other tasks. Executors typically move the current task to the back of their execution queue.